src/evict/evict_lru.c

```diff
@@ -667,6 +669,29 @@ __evict_update_work(WT_SESSION_IMPL *session)
         LF_CLR(WT_CACHE_EVICT_CLEAN | WT_CACHE_EVICT_CLEAN_HARD);
     }
 
+    /*
+     * Clear all non-candidate entries from the eviction queues if the eviction server is about to
+     * go idle, otherwise an under-loaded system can end up in a state where urgent eviction of
+     * pages can be blocked because non-candidate entries are already in the eviction queues, but
+     * are never being visited.
+     */
+    if (F_ISSET(cache, WT_CACHE_EVICT_ALL) && !LF_ISSET(WT_CACHE_EVICT_ALL)) {
+        /* clear all entries from eviction queues (except the urgent queue). */
+        __wt_spin_lock(session, &cache->evict_queue_lock);
+
+        for (q = 0; q < WT_EVICT_QUEUE_MAX - 1; q++) {
+            queue = &cache->evict_queues[q];
+            __wt_spin_lock(session, &queue->evict_lock);
+
+            while (queue->evict_candidates < queue->evict_entries)
```

Comment:
I'd clear all entries - it's reasonable to drain candidates before stopping regular eviction IMO.

That might make it a bit slower when eviction restarts, but if we don't do that it might not solve the issue.

Reply:
Sure @agorrod, removing the candidate pages has improved the performance of the repro in the WT ticket (I will make the change soon). It also seems to have worked for the reproducer posted in the Server ticket, although it doesn't seem to have resolved the issue with mass_regression_delete (this has different symptoms anyway). Also, we are not entirely sure if this will help the HELP tickets that had almost similar symptoms.

I noticed that you posted a comment on the WT ticket with a revised solution that includes calling `__evict_clear_all_walks`. Can you clarify if there would be any benefits to starting the eviction walk from the root next time the eviction server runs?

Reply:
I don't expect the clear_walks part of the change will alter the performance profile - it just seemed like a sensible thing to do in combination with clearing the candidate lists if the eviction server is about to be put on pause.

I thought about mentioning it in the code review, but decided it's fine to just solve the problem at hand here and not extend further.

Regards the `mass_regression_delete` - did the prior change fix that regression? It doesn't seem obviously related to me.

Reply:
Thanks, Alex, the changes requested are updated in [66c8ca3](https://github.com/wiredtiger/wiredtiger/pull/8755/commits/66c8ca3a6384278b2c7bbaf6e9bdc933e8fa5ed5). 

No prior change did not fix the regression.